### PR TITLE
PXE:1043  Dos and Donts Component

### PR DESCRIPTION
**Pull request by smcmullan-r7**
## Description (brief description on what changes have been added)

- Added dos and dont's block variant to tabs component 
- Added check, cross, check-bordered and cross-bordered svgs to block variant 
- Updated SCSS
- Updated Exporter JSON file

JIRA: https://issues.corp.rapid7.com/browse/PXE-1043

## Screenshots (if appropriate):
Tested via test design system and test-documentation-exporter.
Screenshot 2022-11-01 at 22 50 26

Screenshot 2022-11-01 at 22 50 36

Screenshot 2022-11-01 at 22 49 42

## Determine your change type:

_When determining the type of change, only the acceptable use of the component or style should be used to classify the effects of the change._

- Is this a MAJOR? _(If you can check any of these, you should mark your PR as a Major)_
  - [ ] How the component is used/consumed has changed? _(API change, etc)_
  - [ ] The component/module name in the [theme mixin](https://github.com/rapid7/rapid7-ui/blob/master/src/scss/base/_utilities.scss#L136) has changed?
  - ~~The name of the styles mixin has changed?~~
  - [ ] The classnames associated with a component have changed?
  - If you've checked any boxes above: [Read This](https://github.com/rapid7/rapid7-ui/blob/master/CONTRIBUTING.md#release-and-deployment).
- Is this a MINOR? _(you haven't checked anything above, of course)_
  - [x] You're adding completely new functionality
  - [ ] The functionality you're adding is **fully** backwards compatible. _(If it isn't, your change is a MAJOR)_
- Is this a PATCH? _(nothing checked above)_
  - [ ] You're fixing existing functionality
  - [ ] The name of the styles mixin has changed?
  - [ ] Your fix is non-breaking
- No release needed
  - [ ] Changes to tests
  - [ ] Changes to Storybook
  - [ ] Changes to documentation / Readme

## Checklist:

- [ ] I have ensured backwards compatibility and have discussed a deprecation plan with the UI Platform Working Group (if your changes would be considered a MAJOR).
- [x] I have clearly communicated any SCSS changes that fundamentally change display and layout as per [this proposal](https://docs.google.com/document/d/1dNYFL_7u2J3wImFGZBkaoEfuavtmX_g7LftmNdyV5qo/edit?usp=sharing).
- [x] I have updated the documentation accordingly (if changes are required).
- [ ] I have added tests to cover my changes. (if required).
- [ ] I have added a new dependency and properly considered whether that dependency should be in `devDependencies` vs `dependencies`. _for further reading: https://docs.npmjs.com/specifying-dependencies-and-devdependencies-in-a-package-json-file_
- [x] All new and existing tests passed.
- [ ] Added/update story book stories.

### CODE REVIEW

File: scss/blocks/tabs.scss [hunk 1]
```diff
@@ -258,8 +263,92 @@
...
+.donts h3::before {
+  content: '';
+  display: block;
+  background-image: $cross-bordered;
+  background-size: cover;
+  background-repeat: no-repeat;
+  height: 30px;
+  width: 30px;
+  position: absolute;
+  left: 0;
```
Comment (msmichova-r7) [inline scss/blocks/tabs.scss]: we could optimise some of this css as there is a bunch of repetition? things that are the same for both blocks can live under a block like this
```
.dos h3::before, .donts h3::before {
    // styling shared across both
}
```
and then the things that differ can have their own blocks like you have above :)

File: scss/blocks/tabs.scss [hunk 2]
```diff
@@ -258,8 +263,92 @@
...
+  display: block;
+  background-image: $cross;
+  background-size: cover;
+  background-repeat: no-repeat;
+  background-position: left top;
+  height: 28px;
+  width: 28px;
+  position: absolute;
+  left: 0;
+}
```
Comment (msmichova-r7) [inline scss/blocks/tabs.scss]: you have a similar block starting on line 284, we can optimise here as well similarly to my suggestion above ⬆️

File: src/page_body/structure/blocks/page_block_tabs.pr [hunk 3]
```diff
@@ -2,7 +2,13 @@
...
 {[ let blockId = block.id.prefixed("T") /]}
 
-{* This block is variant enabled, fetch it *}
+{* Logos *}
+{[ let checkBordered = assetUrl("logos/check-bordered.svg", ds.documentationDomain()) /]}
+{[ let check = assetUrl("logos/check.svg", ds.documentationDomain()) /]}
+{[ let crossBordered = assetUrl("logos/cross-bordered.svg", ds.documentationDomain()) /]}
+{[ let cross = assetUrl("logos/cross.svg", ds.documentationDomain()) /]}
+
+{* {This block is variant enabled, fetch it} *}
```
Comment (msmichova-r7) [inline src/page_body/structure/blocks/page_block_tabs.pr:12]: remove curly brackets from around the comment :)

File: src/page_body/structure/blocks/page_block_tabs.pr [hunk 4]
```diff
@@ -65,14 +71,27 @@
...
+        {[ let id = tab.id.prefixed("T") /]}
+      <div class="col-sm {{ index === "0" ? "dos" : "donts" }}">
         {[ for child in tab.children ]}
           {[ inject "page_body_structure_block" context child /]}
         {[/]}
       </div>
     {[/]}
   </div>
 
+
```
Comment (msmichova-r7) [inline src/page_body/structure/blocks/page_block_tabs.pr]: remove space

File: src/page_body/structure/blocks/page_block_tabs.pr [hunk 5]
```diff
@@ -65,14 +71,27 @@
...
         {[ let id = tab.id.prefixed("T") /]}
-      <div class="col-sm">
+      <div class="col-sm" >
+        {[ for child in tab.children ]}
+          {[ inject "page_body_structure_block" context child /]}
+        {[/]}
+      </div>
+    {[/]}
+  </div>
+{[ elseif (variant === "dos-donts") ]}
```
Comment (msmichova-r7) [inline src/page_body/structure/blocks/page_block_tabs.pr]: 🔥

File: scss/blocks/tabs.scss [hunk 6]
```diff
@@ -147,7 +149,6 @@
 }
 
 .docs-content .tab-pane > :last-child {
-  padding-bottom: 0px; // originally 16px, but it was breaking the last block
```
Comment (msmichova-r7) [inline scss/blocks/tabs.scss]: do we need to remove this?

File: scss/blocks/tabs.scss [hunk 7]
```diff
@@ -228,13 +229,17 @@
   background-color: transparent;
   color: var(--colorInkPrimary);
   border-top: 1px solid var(--colorBorderColorSubtle);
-  border-bottom: 1px solid var(--colorBorderColorSubtle);
+  border-bottom: 1px solid transparent;
```
Comment (msmichova-r7) [inline scss/blocks/tabs.scss]: let's leave this as it was there originally unless we have a reason to change :)

File: scss/blocks/tabs.scss [hunk 8]
```diff
@@ -228,13 +229,17 @@
...
 }
 
 .accordion-section a.collapsed .accordion-header:hover {
   background-color: var(--colorBackgroundSecondary);
 }
 
+.accordion-section:last-of-type a.collapsed .accordion-header {
+  border-bottom: 0px;
+}
+
```
Comment (msmichova-r7) [inline scss/blocks/tabs.scss]: where is this from? I think we should edit Supernova's code as little as we can :)

File: scss/blocks/tabs.scss [hunk 9]
```diff
@@ -258,8 +263,92 @@
   margin-bottom: 0px;
 }
 
-.accordion-body > :nth-child(2) {
+.accordion-body > :nth-child(2),
+.tabs-as-columns .col-sm > :nth-child(2) {
   // 2nd child, because the first one is anchor
-  padding-top: 0px;
+  padding-top: 10px;
```
Comment (msmichova-r7) [inline scss/blocks/tabs.scss]: same here, if we don't need to edit Supernova's code, let's not touch it 👍

File: exporter.json [hunk 10]
```diff
@@ -481,7 +481,12 @@
         {
           "name": "Accordion",
           "key": "accordion"
+        },
+        {
+          "name": "Dos & Donts",
```
Comment (msmichova-r7) [inline exporter.json]: missing apostrophe on Don'ts :)

File: scss/blocks/tabs.scss [hunk 11]
```diff
@@ -120,7 +122,7 @@
   background: var(--colorBackgroundPrimary);
   border: 1px solid var(--colorBorderColorSubtle);
   border-radius: var(--borderRadiusLarge);
-  padding: 16px 24px;
+  padding: 24px;
```
Comment (msmichova-r7) [inline scss/blocks/tabs.scss]: where's this change from?

**Review by msmichova-r7 — CHANGES_REQUESTED** [review]
A few small changes, but in summary:
1. Let's optimise the CSS as much as we can
2. We should only be changing the Dos and Don'ts styling, we don't want to alter any original styling that could have downstream effect. As we don't have any visual tests, we are trying to be as careful as we can with what we edit :)
3. Spelling and spacing. Nothing to worry about, we could try and have a linter set up for this in future 👍
4. Let's also update the Contributing section in the Supernova editor here https://cloud.supernova.io/ws/rapid7-x/ds/10310-rapid7-x/latest/documentation/editor with instructions on how to use the new component

But this is brilliant work, thank you for your hard work on this 🔥🔥🔥

File: scss/blocks/tabs.scss (hunk 8 above)
Comment (smcmullan-r7) [reply scss/blocks/tabs.scss]: I'm not sure where thats came from, i had no reason to touch any of the accordion scss?

**Review by msmichova-r7 — APPROVED** [review]
🔥

**Pull request merged**